### PR TITLE
refactor(rslint_parser): Restructure simple statements

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Renames and restructures the following statements as we agreed to in our new JS grammar (#1719):

* BlockStatement: Rename node and rename `stmts` child to `statements`
* DebuggerStatement: Rename node
* EmptyStatement: Rename node
* ExpressionStatement: Rename node, `expr` child to `expression`, and add optional trailing semicolon
* ReturnStatement: Rename node, and rename 'value' child to 'argument' 
* LabeledStatement: Rename node, change type of `label` child to a token, and rename `stmt` child to `body`
* WithStatement: 
  * Rename node
  * inline "condition" (it's not a condition)
  * rename `condition` to `object`
  * rename `cons` to `body`
  * Add parser test
  
## Test Plan

Added new parser tests for statements that didn't have a test to this point. Verified that the coverage doesn't change.

## How to Review
It's probably easiest to look at the commits separately

### CODE REVIEW

File: crates/rslint_parser/src/tests.rs [hunk 1]
```diff
@@ -43,8 +43,27 @@ fn test_data_dir() -> PathBuf {
...
-fn try_parse(path: &str, text: &str) -> Parse<JsModule> {
-	let res = catch_unwind(|| parse_module(text, 0));
+struct ParseResult {
+	root: SyntaxNode,
+	errors: Vec<ParserError>,
+}
+
+fn try_parse(path: &str, text: &str) -> ParseResult {
+	let res = catch_unwind(|| {
+		if text.contains("// SCRIPT") {
```
Comment (ematipico) [inline crates/rslint_parser/src/tests.rs:53]: Maybe we should leave a comment here. Or somewhere in the contributing section, testing section... I don't know. Just so we don't lose it

File: crates/rslint_parser/src/syntax/stmt.rs [hunk 2]
```diff
@@ -656,10 +656,14 @@ pub fn if_stmt(p: &mut Parser) -> CompletedMarker {
 pub fn with_stmt(p: &mut Parser) -> CompletedMarker {
 	let m = p.start();
 	p.expect(T![with]);
-	condition(p);
+
+	p.expect(T!['(']);
+	expr(p);
+	p.expect(T![')']);
```
Comment (ematipico) [inline crates/rslint_parser/src/syntax/stmt.rs:662]: Should we add a test case? Or, do we have it already?

```js
while (a === true
```

File: xtask/js.ungram [hunk 3]
```diff
@@ -150,14 +151,21 @@ ContinueStmt = 'continue' name_ref: NameRef ';'?
...
+
+JsWithStatement =
+    'with' '(' object: JsAnyExpression ')'
+    body: JsAnyStatement
+
 
-// with statement
-WithStmt = 'with' condition:Condition cons:JsAnyStatement
+JsLabeledStatement =
+    label: 'js_identifier' ':'
```
Comment (ematipico) [inline xtask/js.ungram:166]: Should have a whole line for `':'`?

File: xtask/js.ungram [hunk 4]
```diff
@@ -178,15 +186,16 @@ ThrowStmt = 'throw' exception:JsAnyExpression ';'
...
 // try..catch..finally statement
 
-TryStmt = 'try' test:BlockStmt handler:CatchClause? finalizer:Finalizer?
+TryStmt = 'try' test:JsBlockStatement handler:CatchClause? finalizer:Finalizer?
 
 // TODO: fix it, the pattern is not mandatory anymore
-CatchClause = 'catch' '(' error:Pattern ')' cons:BlockStmt
-Finalizer = 'finally' cons:BlockStmt
+CatchClause = 'catch' '(' error:Pattern ')' cons:JsBlockStatement
+Finalizer = 'finally' cons:JsBlockStatement
```
Comment (ematipico) [inline xtask/js.ungram:193]:
```suggestion
CatchClause = 'catch' '(' error:Pattern ')' cons: JsBlockStatement
Finalizer = 'finally' cons: JsBlockStatement
```

File: xtask/js.ungram [hunk 5]
```diff
@@ -178,15 +186,16 @@ ThrowStmt = 'throw' exception:JsAnyExpression ';'
 
 // try..catch..finally statement
 
-TryStmt = 'try' test:BlockStmt handler:CatchClause? finalizer:Finalizer?
+TryStmt = 'try' test:JsBlockStatement handler:CatchClause? finalizer:Finalizer?
```
Comment (ematipico) [inline xtask/js.ungram:189]:
```suggestion
TryStmt = 'try' test: JsBlockStatement handler:CatchClause? finalizer:Finalizer?
```

File: xtask/js.ungram [hunk 6]
```diff
@@ -277,7 +286,7 @@ FnExpr =
 	parameters:ArgList
 	':'?
 	return_type:TsType?
-	body:BlockStmt
+	body:JsBlockStatement
```
Comment (ematipico) [inline xtask/js.ungram:289]:
```suggestion
	body: JsBlockStatement
```

File: xtask/js.ungram [hunk 7]
```diff
@@ -326,7 +335,7 @@ Constructor =
 	name:PropName
 	type_params:TsTypeParams?
 	parameters:ParameterList
-	body:BlockStmt
+	body:JsBlockStatement
```
Comment (ematipico) [inline xtask/js.ungram:338]:
```suggestion
	body: JsBlockStatement
```

File: xtask/js.ungram [hunk 8]
```diff
@@ -529,9 +538,9 @@ IdentProp = name: Name
 // MISCELLANEOUS
 ///////////////
 
-Getter =  'get' key:PropName parameters:ParameterList  body:BlockStmt
+Getter =  'get' key:PropName parameters:ParameterList  body:JsBlockStatement
```
Comment (ematipico) [inline xtask/js.ungram:541]:
```suggestion
Getter =  'get' key:PropName parameters:ParameterList  body: JsBlockStatement
```

File: xtask/js.ungram [hunk 9]
```diff
@@ -529,9 +538,9 @@ IdentProp = name: Name
...
 
-Setter = 'set' key:PropName parameters:ParameterList  body:BlockStmt
+Setter = 'set' key:PropName parameters:ParameterList  body:JsBlockStatement
```
Comment (ematipico) [inline xtask/js.ungram:543]:
```suggestion
Setter = 'set' key:PropName parameters:ParameterList  body: JsBlockStatement
```

File: xtask/js.ungram [hunk 10]
```diff
@@ -542,7 +551,7 @@ Method =
 	parameters:ParameterList
 	':'?
 	return_type:TsType?
-	body:BlockStmt
+	body:JsBlockStatement
```
Comment (ematipico) [inline xtask/js.ungram:554]:
```suggestion
	body: JsBlockStatement
```

File: xtask/js.ungram [hunk 11]
```diff
@@ -577,7 +586,7 @@ FnDecl =
...
```
Comment (ematipico) [inline xtask/js.ungram:589]:
```suggestion
	body: JsBlockStatement
```

**Review by ematipico — APPROVED** [review]

File: crates/rslint_parser/src/syntax/stmt.rs (hunk 2 above)
Comment (MichaReiser) [reply crates/rslint_parser/src/syntax/stmt.rs:662]: I don't think there's any for the `with` statement. I don't plan to add it in this PR. I think we should do the same exercise again and go through each node and verify the error recovery implementation.

File: xtask/js.ungram (hunk 5 above)
Comment (MichaReiser) [reply xtask/js.ungram:189]: I'll change the formatting of each node when I change its name & structure. I want to keep the changes in each PR to a minimum.

**Pull request merged**